### ISSUE TITLE
[FR] rules based on frontmatter fields

### ISSUE THREAD

**Issue by andrewcrook** [farux/obsidian-auto-note-mover#25]
-  add type for frontmatter
- use a front-matter field name
- use regex to test value of field

<img width="745" alt="image" src="https://user-images.githubusercontent.com/803618/169694522-51394a5a-a6f9-4b9a-8653-ba1be09acaa2.png">

other possible scenarios 
- option for field not found
- field without value

**Comment by crazymedguy** [comment]
This would be a great approach to using the DB folder plugin

**Comment by isle9** [comment]
> This would be a great approach to using the DB folder plugin

Yeah, I can't add filters based on tags because the db note gets moved.

It would be great if dataview fields were also included, since I use those instead.

**Comment by ashishbhateja** [comment]
I am also looking forward to this enhancement. Any update on this request?

**Comment by ArneNostitz** [comment]
Also would really love if inline fields would work as trigger

**Comment by prashant2402** [comment]
> * add type for frontmatter
> * use a front-matter field name
> * use regex to test value of field
> 
> <img alt="image" width="745" src="https://user-images.githubusercontent.com/803618/169694522-51394a5a-a6f9-4b9a-8653-ba1be09acaa2.png">
> 
> other possible scenarios
> 
> * option for field not found
> * field without value

I have now created a [PR ](https://github.com/farux/obsidian-auto-note-mover/pull/53)with a simple frontmatter property check feature. Tested locally and works well. You can use multiple rules if you want to match an array of frontmatter property values.
![Screenshot 2023-05-27 112259](https://github.com/farux/obsidian-auto-note-mover/assets/1729202/0d14b68e-12b0-40c8-97f7-15eeef16e2d7)
